crates/vault-registry/src/lib.rs

```diff
@@ -383,11 +394,51 @@ impl<T: Config> Module<T> {
         <LiquidationVault<T>>::get()
     }
 
+    pub fn increase_backing_collateral(vault_id: &T::AccountId, amount: DOT<T>) -> DispatchResult {
+        ensure!(
+            Self::is_nomination_operator(vault_id),
+            Error::<T>::VaultIsNotANominationOperator
+        );
+        let mut vault = Self::get_active_rich_vault_from_id(&vault_id)?;
+        vault.increase_backing_collateral(amount)
+    }
+
+    pub fn decrease_backing_collateral(vault_id: &T::AccountId, amount: DOT<T>) -> DispatchResult {
+        ensure!(
+            Self::is_nomination_operator(vault_id),
+            Error::<T>::VaultIsNotANominationOperator
+        );
+        let mut vault = Self::get_active_rich_vault_from_id(&vault_id)?;
+        vault.decrease_backing_collateral(amount)
+    }
+
+    /// Locks `amount` DOT to be used for backing PolkaBTC
+    ///
+    /// # Arguments
+    /// * `vault_id` - the id of the vault from which to increase to-be-issued tokens
+    /// * `amount` - the amount of DOT to be locked
     pub fn try_lock_additional_collateral(vault_id: &T::AccountId, amount: DOT<T>) -> DispatchResult {
+        Self::try_lock_additional_collateral_from_address(vault_id, amount, vault_id)
+    }
+
+    pub fn try_lock_additional_collateral_from_address(
+        vault_id: &T::AccountId,
+        amount: DOT<T>,
+        depositor_id: &T::AccountId,
+    ) -> DispatchResult {
         let mut vault = Self::get_active_rich_vault_from_id(vault_id)?;
 
         // will fail if free_balance is insufficient
-        ext::collateral::lock::<T>(vault_id, amount)?;
+        ext::collateral::lock::<T>(depositor_id, amount)?;
+        if !vault_id.eq(depositor_id) {
+            ext::collateral::repatriate_reserved::<T>(
```

Comment:
Can't we use the `slash_collateral` function?

Reply:
From the docs of `slash_collateral`: "Deducts up to `value` from reserved balance of `who`. This function cannot fail". 

We need this function to fail if the depositor does not have enough funds to make the deposit, as hinted by the comment above (`// will fail if free_balance is insufficient`). I can be more explicit in the comment

Reply:
I mean the `slash_collateral` function that is defined in `vault_registry`. It's a function that can transfer from any type of DOT balance to any other

Reply:
I updated. We may want to change the name of `slash_collateral` in the future since its name no longer reflects what it does